lib/parliament/request/base_request.rb

```diff
@@ -1,78 +1,83 @@
       #
-      #   Parliament::Request.new(headers: '{ 'Accept' => 'Test' }).headers #=> { 'Accept' => 'Test' }
+      #   Parliament::Request::BaseRequest.new(headers: '{ 'Accept' => 'Test' }).headers #=> { 'Accept' => 'Test' }
       #
       # @example Overriding the headers via a parameter
-      #   Parliament::Request.headers = { 'Accept' => 'Test' }
+      #   Parliament::Request::BaseRequest.headers = { 'Accept' => 'Test' }
       #
-      #   Parliament::Request.new(headers: '{ 'Accept' => 'Test2' }).headers #=> { 'Accept' => 'Test2' }
+      #   Parliament::Request::BaseRequest.new(headers: '{ 'Accept' => 'Test2' }).headers #=> { 'Accept' => 'Test2' }
       #
       # @param [String] base_url the base url of our api. (expected: http://example.com - without the trailing slash).
       # @param [Hash] headers the headers being sent in the request.
-      def initialize(base_url: nil, headers: nil, builder: nil)
+      # @param [Parliament::Builder] builder the builder to use in order to build a response.
+      # @params [Module] decorators the decorator module to use in order to provide possible alias methods for any objects created by the builder.
+      def initialize(base_url: nil, headers: nil, builder: nil, decorators: nil)
         @base_url = base_url || self.class.base_url
         @headers = headers || self.class.headers || {}
         @builder = builder || Parliament::Builder::BaseResponseBuilder
+        @decorators = decorators
       end
 
-      # Using our url built via #method_missing, make a HTTP GET request and process results into a response.
+      # Makes an HTTP GET request and process results into a response.
       #
       # @example HTTP GET request
-      #   request = Parliament::Request.new(base_url: 'http://example.com')
-      #
-      #   # url: http://example.com/people/123456
-      #   response = request.people('123456').get #=> #<Parliament::Response ...>
+      #   request = Parliament::Request::BaseRequest.new(base_url: 'http://example.com/people/123'
+      #   response = request.get #=> #<Parliament::Response::BaseResponse ...>
       #
       # @example HTTP GET request with URI encoded form values
-      #   request = Parliament::Request.new(base_url: 'http://example.com')
-      #
-      #   # url: http://example.com/people/current?limit=10&page=4&lang=en-gb
-      #   response = request.people.current.get({ limit: 10, page: 4, lang: 'en-gb' }) #=> #<Parliament::Response ...>
+      #   request = Parliament::Request.new(base_url: 'http://example.com/people/current')
+      #   response = request.get({ limit: 10, page: 4, lang: 'en-gb' }) #=> #<Parliament::Response::BaseResponse ...>
       #
       # @raise [Parliament::ServerError] when the server responds with a 5xx status code.
       # @raise [Parliament::ClientError] when the server responds with a 4xx status code.
       # @raise [Parliament::NoContentResponseError] when the server responds with a 204 status code.
```

Comment:
We should look at this - the new API no longer returns 204 status codes, only 200. This error can stay but it would be more along the lines of 'when the response body is empty' - it may also be specific to the Ntriple or URL request side of things?